### PR TITLE
Improve documentation for the DatasetsMaker

### PR DESCRIPTION
**Pull request by QRemy**
Improve documentation for the `DatasetsMaker`:
-extend docsting about the cutout options
-add an example in the makers notebook

### CODE REVIEW

File: docs/tutorials/api/makers.ipynb [hunk 1]
```diff
@@ -317,7 +318,22 @@
     "per observation into a larger dataset. During the stacking the safe data\n",
     "range mask (`MapDataset.mask_safe`) is applied by setting data outside to\n",
     "zero, then data is added to the larger map dataset. To stack multiple\n",
-    "observations, the larger dataset must be created first."
+    "observations, the larger dataset must be created first.\n",
+    "\n",
+    "The data reduction loop shown above can be done throught the `DatasetsMaker`.  Its `stack_datasets` option offers the possibily to stack or not the output datasets, and the `n_jobs` option allow to use multiple processes on run:"
```
Comment (registerrier) [inline docs/tutorials/api/makers.ipynb:323]: I would add in bold face that the order in which you pass the makers is important

File: docs/tutorials/api/makers.ipynb [hunk 2]
```diff
@@ -287,21 +288,21 @@
...
     ")\n",
     "\n",
-    "maker = MapDatasetMaker()\n",
+    "dataset_maker = MapDatasetMaker()\n",
     "safe_mask_maker = SafeMaskMaker(\n",
     "    methods=[\"aeff-default\", \"offset-max\"], offset_max=\"3 deg\"\n",
     ")\n",
     "\n",
-    "stacked = MapDataset.create(geom)\n",
+    "global_dataset = MapDataset.create(geom)\n",
```
Comment (registerrier) [inline docs/tutorials/api/makers.ipynb:296]: why the change to `global_dataset`? `stacked` is quite explicit no?

File: gammapy/makers/reduce.py [hunk 3]
```diff
@@ -22,14 +22,18 @@ class DatasetsMaker(Maker):
...
-        Stack into reference dataset or not
+        If True stack into the reference dataset (see `run` method arguments).
     n_jobs : int
-        Number of processes to run in parallel
-        Default is None
-    cutout_mode : str
-        Cutout mode. Default is "partial"
-    cutout_width : str or `~astropy.coordinates.Angle`,
-        Cutout width. Default is None, If Default is determined
+        Number of processes to run in parallel. Default is None.
```
Comment (registerrier) [inline gammapy/makers/reduce.py:27]: default should be 1 no?

**Review by registerrier — APPROVED** [review]
Thanks @QRemy ! This looks good.

I have left a few comments inline.

File: gammapy/makers/reduce.py (hunk 3 above)
Comment (QRemy) [reply gammapy/makers/reduce.py]: yes, I also changed the TSMapEstimator for consistency

File: docs/tutorials/api/makers.ipynb (hunk 1 above)
Comment (QRemy) [reply docs/tutorials/api/makers.ipynb]: added

File: docs/tutorials/api/makers.ipynb (hunk 2 above)
Comment (QRemy) [reply docs/tutorials/api/makers.ipynb]: changed back to stacked

File: gammapy/makers/reduce.py (hunk 3 above)
Comment (adonath) [reply gammapy/makers/reduce.py]: @registerrier No, the default of `None` was introduced on purpose and I think it should remain! The difference is that `n_jobs=None` implements a second path which avoids into calling into `multiprocessing` at all. The reason was that we have seen issues with `multiprocessing` in the past. See e.g. https://github.com/gammapy/gammapy/issues/2453. That's why the default should remain `None` (i.e. not calling into `multiprocessing` at all), optionally users can then use the parallelisation, but it is not guaranteed to work in all situations!

File: gammapy/estimators/map/ts.py [hunk 4]
```diff
@@ -386,13 +386,12 @@ def estimate_flux_map(self, dataset):
...
-        if self.n_jobs is None:
-            results = list(map(wrap, positions))
-        else:
+        if self.n_jobs > 1 :
             with Pool(processes=self.n_jobs) as pool:
                 log.info("Using {} jobs to compute TS map.".format(self.n_jobs))
                 results = pool.map(wrap, positions)
-
+        else:
+            results = list(map(wrap, positions))
```
Comment (QRemy) [inline gammapy/estimators/map/ts.py:394]: @adonath this is equivalent to checking if n_jobs is None

File: gammapy/makers/reduce.py (hunk 3 above)
Comment (QRemy) [reply gammapy/makers/reduce.py]: running multiprocessing only if n_jobs > 1 is equivalent, cf the change below

**Review by registerrier — APPROVED** [review]
Thanks @QRemy ! This looks good.

Regarding `n_jobs` argument, it is probably more clear that `n_job=1` means one process. Otherwise, one needs to explain that `n_job=1` is one process in multiprocessing mode. This might be useful to debug I suppose, but probably confusing to users. Is the solution OK with you @adonath ?

**Comment by registerrier** [comment]
@adonath, do you think the `n_jobs` argument should be `None` for no multiprocessing, or is the current solution with default at 1 and multiprocessing called only for `n_jobs >=2` fine with you?

File: gammapy/makers/reduce.py [hunk 5]
```diff
@@ -22,14 +22,18 @@ class DatasetsMaker(Maker):
...
-        Cutout width. Default is None, If Default is determined
+        Number of processes to run in parallel. Default is 1.
+    cutout_mode : {'trim', 'partial', 'strict'}
+        Used only to cutout the refrence MapDataset around each processed observation.
+        Mode is an option for Cutout2D, for details see `~astropy.nddata.utils.Cutout2D`.
+        Default is "trim"..
+    cutout_width : tuple of `~astropy.coordinates.Angle`
+        Angular sizes of the region in (lon, lat) in that specific order.
+        If only one value is passed, a square region is extracted.
+        If None it is set to 2*offset-max if this option is given to the SafeMaskmaker
```
Comment (registerrier) [inline gammapy/makers/reduce.py:35]: 2*offset_max might be clearer

File: gammapy/makers/reduce.py [hunk 6]
```diff
@@ -22,14 +22,18 @@ class DatasetsMaker(Maker):
...
-        Default is None
-    cutout_mode : str
-        Cutout mode. Default is "trim"
-    cutout_width : str or `~astropy.coordinates.Angle`,
-        Cutout width. Default is None, If Default is determined
+        Number of processes to run in parallel. Default is 1.
+    cutout_mode : {'trim', 'partial', 'strict'}
+        Used only to cutout the refrence MapDataset around each processed observation.
+        Mode is an option for Cutout2D, for details see `~astropy.nddata.utils.Cutout2D`.
+        Default is "trim"..
```
Comment (registerrier) [inline gammapy/makers/reduce.py:31]: Remove last dot.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy ! I have left two nitpick comments.

The CI failure is unrelated (due to a filename change in gammapy-data)

**Comment by adonath** [comment]
@QRemy and @registerrier I just looked into the `n_jobs` argument again and I now remember that I followed `scikit-learn` when first introducing the argument. `scikit-learn` specifies `n_jobs=None` by default, the reason is given [here](https://scikit-learn.org/stable/glossary.html#term-n_jobs). So `scikit-learn` has backend independent parallelization support and in some cases the number of jobs are configured "outside", like:
```python3
from ray.util.joblib import register_ray  
register_ray()

with parallel_backend("ray", n_jobs=10):  
    print(Parallel()(delayed(neg)(i + 1) for i in range(5)))

```
In which case the `n_jobs=None` corresponds to number of jobs defined in the context, otherwise it defaults to one job. We don't have anything like this in Gammapy (yet), but I think the need for an additional placeholder value is probably still justified for future developments.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy ! Fine with the None default.

**Comment by registerrier** [comment]
CI fail is unrelated. Merging.
